### PR TITLE
Switch busybox over to co-maintainership

### PR DESCRIPTION
**Pull request by tianon**
Also, update it with a new version built directly from upstream's sources (instead of relying on buildroot to update their embedded copy of busybox).

cc @jpetazzo

### CODE REVIEW

**Comment by tianon** [comment]
Build test of #1058; 6c5ea7b84c35f58818be3cb9f46e76d2f009df69 (`busybox`):

```console
$ url="https://raw.githubusercontent.com/docker-library/official-images/6c5ea7b84c35f58818be3cb9f46e76d2f009df69/library/busybox"
$ bashbrew build "$url"
Cloning busybox (git://github.com/docker-library/busybox) ...
Processing busybox:ubuntu-1.21.0 ...
Processing busybox:ubuntu-1.21 ...
Processing busybox:ubuntu-1 ...
Processing busybox:ubuntu ...
Processing busybox:1.23.2 ...
Processing busybox:1.23 ...
Processing busybox:1 ...
Processing busybox:latest ...
$ bashbrew list --uniq "$url" | xargs test/run.sh
testing busybox:ubuntu-1.21.0
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed
testing busybox:1.23.2
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...warning: garbage password detected for 'mail': 'x'
warning: garbage password detected for 'www-data': 'x'
warning: garbage password detected for 'sys': 'x'
passed
	'override-cmd' [4/4]...passed
```

**Comment by tianon** [comment]
@jpetazzo do you think we should switch the `ubuntu` version to pull from wily instead of trusty so that it's for-sure the latest and greatest Ubuntu has to offer?

**Comment by tianon** [comment]
That would take us to [`1.22.0`](http://packages.ubuntu.com/wily/busybox-static) (from [`1.21.0`](http://packages.ubuntu.com/trusty/busybox-static)).

**Comment by tianon** [comment]
Maybe get this out first and then consider that separately.

**Comment by jpetazzo** [comment]
Ah, 1.22 is always better than 1.21, but it's not 1.23...

I wouldn't mind dropping the ubuntu builder altogether if it makes maintenance easier.

(In the beginning, I had two incentives to keep it around: it was an alternative to buildroot; and it was faster... But I don't know if those criteria should still be a concern!)

**Comment by tianon** [comment]
Heh, the tarballs for it build fast enough that I'm not too worried about it for now.  I'm fairly sure it's built against glibc too, so IMO it's still worth keeping around for a while longer (especially since the burden of it is low).  Let's look at moving it to the Wily version in a later PR though. :+1:

File: library/busybox [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
 # maintainer: Jérôme Petazzoni <jerome@docker.com> (@jpetazzo)
 
-buildroot-2013.08.1: git://github.com/jpetazzo/docker-busybox@220a689ce359914af3e08a698d1d74ec7aa0a444
-buildroot-2014.02: git://github.com/jpetazzo/docker-busybox@91641afe424df5e838bac254d43e09f051ab8c3e
-latest: git://github.com/jpetazzo/docker-busybox@91641afe424df5e838bac254d43e09f051ab8c3e
-ubuntu-12.04: git://github.com/jpetazzo/docker-busybox@4f6cb64c3b3255c58021dc75100da0088796a108
-ubuntu-14.04: git://github.com/jpetazzo/docker-busybox@ca435164f45c40d761fad9ef9b5a76a6ba0d5f1a
+ubuntu-1.21.0: git://github.com/docker-library/busybox@9018eeb5d4be894da7fa94d21346d37dbd8ee847 ubuntu
+ubuntu-1.21: git://github.com/docker-library/busybox@9018eeb5d4be894da7fa94d21346d37dbd8ee847 ubuntu
+ubuntu-1: git://github.com/docker-library/busybox@9018eeb5d4be894da7fa94d21346d37dbd8ee847 ubuntu
```
Comment (yosifkit) [inline library/busybox]: These should be swapped to be `version-ubuntu`, since that is the version of the software and not the version of Ubuntu.

Comment (tianon) [inline library/busybox]: Fixed!

**Comment by tianon** [comment]
Build test of #1058; 77200a03e1e7af7c7a79f015db7cc77d4eb4cd03 (`busybox`):

```console
$ url="https://raw.githubusercontent.com/docker-library/official-images/77200a03e1e7af7c7a79f015db7cc77d4eb4cd03/library/busybox"
$ bashbrew build "$url"
Fetching busybox (git://github.com/docker-library/busybox) ...
Processing busybox:1.21.0-ubuntu ...
Processing busybox:1.21-ubuntu ...
Processing busybox:1-ubuntu ...
Processing busybox:ubuntu ...
Processing busybox:1.23.2 ...
Processing busybox:1.23 ...
Processing busybox:1 ...
Processing busybox:latest ...
$ bashbrew list --uniq "$url" | xargs test/run.sh
testing busybox:1.21.0-ubuntu
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed
testing busybox:1.23.2
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...warning: garbage password detected for 'mail': 'x'
warning: garbage password detected for 'www-data': 'x'
warning: garbage password detected for 'sys': 'x'
passed
	'override-cmd' [4/4]...passed
```

**Comment by yosifkit** [comment]
LGTM

**Pull request merged**